addons/repair/models/repair.py

```diff
@@ -172,11 +174,35 @@ def _default_picking_type_id(self):
         'Allowed to Reserve Production', compute='_compute_unreserve_visible',
         help='Technical field to check when we can reserve quantities')
 
+    @api.depends('product_id', 'picking_id', 'lot_id')
+    def _compute_product_qty(self):
+        for repair in self:
+            if repair.picking_id:
+                if repair.tracking in ['serial', 'lot'] and repair.lot_id:
+                    repair.product_qty = repair.picking_id.move_line_ids.filtered(lambda m: m.product_id == repair.product_id and m.lot_id == repair.lot_id).quantity
+                else:
+                    repair.product_qty = repair.picking_id.move_ids.filtered(lambda m: m.product_id == repair.product_id).quantity
```

Comment:
An error can happen if the `filtered` found multiple records (can happen if you have multiple moves for the same product, or multiple move lines for the same lot)
```suggestion
                    lot_move_lines = repair.picking_id.move_line_ids.filtered(lambda m: m.product_id == repair.product_id and m.lot_id == repair.lot_id)
                    repair.product_qty = sum(lot_move_lines.mapped('quantity'))
                else:
                    product_moves = repair.picking_id.move_ids.filtered(lambda m: m.product_id == repair.product_id)
                    repair.product_qty = sum(product_moves.mapped('quantity'))
```